pkg/kubelet/kubelet_pods.go

```diff
@@ -1588,21 +1593,40 @@ func (kl *Kubelet) convertStatusToAPIStatus(pod *v1.Pod, podStatus *kubecontaine
 		validPrimaryIP = utilnet.IsIPv6String
 		validSecondaryIP = utilnet.IsIPv4String
 	}
-	for _, ip := range podStatus.IPs {
+	for _, ip := range podIPs {
 		if validPrimaryIP(ip) {
-			podIPs = append(podIPs, v1.PodIP{IP: ip})
+			ips = append(ips, ip)
 			break
 		}
 	}
-	for _, ip := range podStatus.IPs {
+	for _, ip := range podIPs {
 		if validSecondaryIP(ip) {
-			podIPs = append(podIPs, v1.PodIP{IP: ip})
+			ips = append(ips, ip)
 			break
 		}
 	}
-	apiPodStatus.PodIPs = podIPs
-	if len(podIPs) > 0 {
-		apiPodStatus.PodIP = podIPs[0].IP
+	return ips
+}
+
+// convertStatusToAPIStatus creates an api PodStatus for the given pod from
+// the given internal pod status.  It is purely transformative and does not
+// alter the kubelet state at all.
+func (kl *Kubelet) convertStatusToAPIStatus(pod *v1.Pod, podStatus *kubecontainer.PodStatus) *v1.PodStatus {
+	var apiPodStatus v1.PodStatus
+
+	// copy pod status IPs to avoid race conditions with PodStatus #102806
+	podIPs := make([]string, len(podStatus.IPs))
+	for j, ip := range podStatus.IPs {
+		podIPs[j] = ip
+	}
+
+	// make podIPs order match node IP family preference #97979
+	podIPs = kl.sortPodIPs(podIPs)
```

Comment:
there is some subtle golang thing that disallow this?

Reply:
I don't know - is something not working?

Reply:
it does work, I just pass the slice as argument and override the same variable with the output, I think is fine, but I just wanted to double check